### PR TITLE
refactor: Remove api -> contexts-config-store dependency by checking Portal existence in the client

### PR DESCRIPTION
**Pull request by gbouv**
## Description:
This will make consuming Kurtosis API much easier

## Is this change user facing?
NO

### CODE REVIEW

File: cli/cli/commands/kurtosis_context/context_switch/switch.go [hunk 1]
```diff
@@ -89,15 +87,20 @@ func run(ctx context.Context, _ *flags.ParsedFlags, args *args.ParsedArgs) error
...
+		if portalDaemonClient != nil {
+			switchContextArg := constructors.NewSwitchContextArgs()
+			if _, err = portalDaemonClient.SwitchContext(ctx, switchContextArg); err != nil {
+				return stacktrace.Propagate(err, "Error switching Kurtosis portal context")
+			}
+		}
+	} else {
+		if store.IsRemote(currentContext) {
+			return stacktrace.NewError("New context is remote but Kurtosis Portal is not reachable locally. " +
+				"Make sure Kurtosis Portal is running before switching to a remote context again.")
```
Comment (h4ck3rk3y) [inline cli/cli/commands/kurtosis_context/context_switch/switch.go:103]: nit - maybe we can tell them how to

File: contexts-config-store/store/contexts_config_store.go [hunk 2]
```diff
@@ -37,3 +38,18 @@ func GetContextsConfigStore() ContextsConfigStore {
 	})
 	return contextsConfigStore
 }
+
+func IsRemote(kurtosisContext *generated.KurtosisContext) bool {
+	var isRemote bool
+	_, _ = golang.Visit[struct{}](kurtosisContext, golang.KurtosisContextVisitor[struct{}]{
```
Comment (h4ck3rk3y) [inline contexts-config-store/store/contexts_config_store.go:44]: the visitor pattern has been quite useful so far; i think I am seeing the 5th implementation of this for remote/local haha

**Review by h4ck3rk3y — APPROVED** [review]

File: cli/cli/commands/kurtosis_context/context_switch/switch.go (hunk 1 above)
Comment (gbouv) [reply cli/cli/commands/kurtosis_context/context_switch/switch.go:103]: We don't have a command for this yet, but that's my next PR. Will add once the command is in :)

File: contexts-config-store/store/contexts_config_store.go (hunk 2 above)
Comment (gbouv) [reply contexts-config-store/store/contexts_config_store.go:44]: visitors are the best :D

**Pull request merged**